src/games/strategy/ui/ProgressDialog.java

```diff
@@ -0,0 +1,75 @@
+package games.strategy.ui;
+
+import java.awt.BorderLayout;
+import java.awt.Color;
+import java.awt.Frame;
+import java.util.Objects;
+
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JProgressBar;
+import javax.swing.border.EmptyBorder;
+import javax.swing.border.LineBorder;
+
+/**
+ * A modal dialog used to display indeterminate progress during an operation.
+ */
+public final class ProgressDialog extends JDialog {
+  private static final long serialVersionUID = -590470596784214914L;
+
+  private final String message;
+
+  /**
+   * Initializes a new instance of the {@code ProgressDialog} class.
+   *
+   * @param owner The {@code Frame} from which the dialog is displayed or {@code null} to use a shared, hidden frame as
+   *        the owner of the dialog.
+   * @param message The progress message; must not be {@code null}.
+   */
+  public ProgressDialog(final Frame owner, final String message) {
+    super(owner, true);
+
+    Objects.requireNonNull(message);
```

Comment:
Chalk that up to laziness.  I didn't even bother to look at the rest of the codebase to see if there were existing libraries being used that provide this functionality.  (I probably would have if it didn't exist in the Java standard library.)  I'll switch to Guava as needed.